app/javascript/components/pages/Feed.js

```diff
@@ -2,46 +2,120 @@ import React from "react"
+          markerLng: 0, 
+          markerPosition: [0,0],
+          markers: []
+        }   
+      }
+    //need this to load markers when switching between components
     componentDidMount() {
+        const { posts } = this.props
         this.props.changeSuccess(false)
+        const provider = new OpenStreetMapProvider();
+        posts.map((post)=>{
+            provider.search({ query: post.location })
+            .then((result) => { 
+                this.setState({markerLat:result[0].y, markerLng:result[0].x})
+                let title = post.title
+                let description = post.description
+                let marker = {
+                    location: [this.state.markerLat, this.state.markerLng], 
+                    title: title, 
+                    description: description
+                }
+                this.setState({markers: [...this.state.markers, marker ]})
+            })
+        })
     }
-
-    render () {
-        const {
-            posts,
-            handleDeletePost,
-            current_user_id
-        } = this.props
-    return (
-      <React.Fragment>
-        <h1>Deed Feed</h1>
-        <div>
-            {posts.map((post) => {
-                return(
-                    <div key={post.id}>
-                        {post.title}
-                        {post.description}
-                        {post.location}
-                        {post.user_id === current_user_id &&
-                            <div>
-                            <Button>
-                                <Link to={`/edit_post/${post.id}`}>Edit Post
-                                </Link>
-                            </Button>
-                            <Button onClick={() => window.confirm("Are you sure you wish to delete post?") && this.props.handleDeletePost(post.id)}>Delete Post</Button>
-                            </div>
-                        }
-                    </div>
-                )
-            })}
-        </div>
-      </React.Fragment>
-    );
+    //used to load markers when the state of posts is changed
+    componentDidUpdate(prevProps) {
+        const { posts } = this.props
+        if(prevProps.posts != posts ){
+            //used to reset state to rerender component
+            this.setState({markers: []})
+            const provider = new OpenStreetMapProvider();
+            posts.map((post)=>{
+                provider.search({ query: post.location })
+                .then((result) => {
+                    this.setState({markerLat:result[0].y, markerLng:result[0].x})
+                    let title = post.title
+                    let description = post.description
+                    let marker = {
+                        location: [this.state.markerLat, this.state.markerLng], 
+                        title: title, 
+                        description: description
+                    }
+                    this.setState({markers: [...this.state.markers, marker ]})
```

Comment:
This looks like a duplicate code to componentDidMount.  Let's pull it out to a function that is called from both places.